### PR TITLE
[Review] Request from 'vlewin' @ 'SUSE/connect/review_150731_zypper_products_search'

### PR DESCRIPTION
**Pull request by vlewin**
Please review the following changes:
  * 0bdb235 fix rubocop offeces
  * c7a3d78 return an array of OpenStruct object
  * 286c2a1 fix rubocop offences
  * e292231 expose Zypper find_products method to migration layer
  * 0ddd3a2 introduce find_products method
  * 43524e3 add fixture for zypper products search

### CODE REVIEW

File: lib/suse/connect/migration.rb [hunk 1]
```diff
@@ -49,8 +48,14 @@ def remove_service(service_name)
           SUSE::Connect::Zypper.remove_service(service_name)
         end
 
+        # Finds the solvable products available on the system
+        # @param [String] identifier e.g. SLES
+        # @return [Array <OpenStruct>] the list of solvable products available on the system
+        def find_products(identifier)
+          # INFO: use block instead of .map(&:to_openstruct) see https://bugs.ruby-lang.org/issues/9786
+          SUSE::Connect::Zypper.find_products(identifier).map {|p| p.to_openstruct }
```
Comment (kalabiyau) [inline lib/suse/connect/migration.rb]: Why not?
```ruby
 SUSE::Connect::Zypper.find_products(identifier).map(&:to_openstruct)
```

Comment (vlewin) [inline lib/suse/connect/migration.rb]: See the INFO: comment above this line

File: lib/suse/connect/zypper.rb [hunk 2]
```diff
@@ -65,6 +65,14 @@ def remove_service(service_name)
           remove_service_credentials(service_name)
         end
 
+        # @param product identifier [String]
+        # Returns an array of hashes of all solvable products
+        def find_products(identifier)
+          zypper_out = call("--xmlout --non-interactive search -s -t product #{identifier}", false)
+          xml_doc = REXML::Document.new(zypper_out, compress_whitespace: [])
+          xml_doc.elements.each('stream/search-result/solvable-list/solvable') {}.map(&:to_hash)
```
Comment (kalabiyau) [inline lib/suse/connect/zypper.rb]: Very strange construction. I will check that one

Comment (kalabiyau) [inline lib/suse/connect/zypper.rb]: IMO notation below
* is way easier to grasp and you don't need to skip yielding
* does not use iterator which is not needed here

```ruby
xml_doc.elements.to_a('stream/search-result/solvable-list/solvable').map(&:to_hash)
```

**Comment by kalabiyau** [comment]
lgtm

**Pull request merged**